apps/balancer-tools/src/app/apr/pool/(components)/HistoricalAPRChart.tsx

```diff
@@ -0,0 +1,56 @@
+import { blueDark } from "@radix-ui/colors";
+
+import { trimTrailingValues } from "#/lib/utils";
+
+import { calculateAPRForPool } from "../../(utils)/calculateRoundAPR";
+import { Round } from "../../(utils)/rounds";
+import HistoricalAPRPlot from "./HistoricalAPRPlot";
+
+export default async function HistoricalAPRChart({
+  roundId,
+  poolId,
+}: {
+  roundId: string;
+  poolId: string;
+}) {
+  const HOVERTEMPLATE = "%{x}<br />%{y:.2f}% APR<extra></extra>";
+  const LAST_ROUND_ID = parseInt(Round.getAllRounds()[0].value);
+  const APRPerRoundCords: { x: string[]; y: number[] } = { x: [], y: [] };
+  for (let index = 1; index < LAST_ROUND_ID; index++) {
+    const { apr } = await calculateAPRForPool({ poolId, roundId: index });
```

Comment:
Why not use a `Promise.all` approach? Would it be too many requests at once? If you need to throttle them you could use https://www.npmjs.com/package/p-throttle

Reply:
yeah, here it is taking 1min20s to load everything ☠️